### PR TITLE
Now Supports Multi-Threading

### PR DESCRIPTION
**Pull request by ryegye24**
The `worker_count` constructor argument is the number of activities which can be performed simultaneously.

Right now I'm using `ThreadPoolExecutor` since it's a little more flexible/lightweight, but I _just_ recently discovered that `ProcessPoolExecutor` now is capable of subverting the GIL?? So I'm toying with the idea of using that.

### CODE REVIEW

File: stepfunctions_activity_worker/activity_worker.py [hunk 1]
```diff
@@ -39,16 +43,8 @@ def _default_config():
         config = botocore.config.Config(read_timeout=70)
         return boto3.client('stepfunctions', config=config)
 
-    def __call__(self):
```
Comment (MaxwellGBrown) [inline stepfunctions_activity_worker/activity_worker.py]: Removal of this is a user-space breaking change; either we'll have to extend `perform_task` to allow people to pass in a task, or we'll have to make this concurrent on the `listen()` call to `perform_task()`

```python
def _poll_for_task(self):
    task = dict()
    while not task.get("taskToken"):
        print("Polling for an activity task.")
        task = self.stepfunctions.get_activity_task(**self.activity_kwargs)
    print("Recieved a task!")
    print(json.dumps(json.loads(task["input"]), indent=4, sort_keys=True))
    return task

def perform_task(self, task=None):
    task = self._poll_for_task() if not task else task
    heartbeat = Heartbeat(...)
    try:
        print(f"Performing {self.activity_name}")
        with heartbeat:
            task_input = json.loads(task["input"])
            output = self.activity_fxn(**task_input)
    except (Exception, KeyboardInterrupt) as error:
        *_, raw_traceback = sys.exc_info()
        # TODO Figure out how to handle 

```

File: stepfunctions_activity_worker/activity_worker.py [hunk 2]
```diff
@@ -68,28 +64,47 @@ def perform_task(self):
...
             raise
+        return output
 
-        print(f"{self.activity_name} is completed!")
-        print(json.dumps(output, indent=4, sort_keys=True))
-        self.stepfunctions.send_task_success(
-            taskToken=task["taskToken"],
-            output=json.dumps(output, sort_keys=True),
-        )
+    def complete_task(self, future, task):
```
Comment (MaxwellGBrown) [inline stepfunctions_activity_worker/activity_worker.py]: This function should be "private"; it doesn't have any use to a user.

```python
def _complete_task(self, future, task):
    ...
```

File: stepfunctions_activity_worker/activity_worker.py [hunk 3]
```diff
@@ -68,28 +64,47 @@ def perform_task(self):
...
+        """Send the task results"""
+        error = future.exception()
+        if error:
```
Comment (MaxwellGBrown) [inline stepfunctions_activity_worker/activity_worker.py]: Error isn't used anywhere except for exception checking.

```
if future.exception():
```

File: stepfunctions_activity_worker/activity_worker.py [hunk 4]
```diff
@@ -68,28 +64,47 @@ def perform_task(self):
...
+            print(f"{self.activity_name} failed!")
+        else:
```
Comment (MaxwellGBrown) [inline stepfunctions_activity_worker/activity_worker.py]: Unnecessary else block; we could probably return in the `if future.exception()` check.

File: stepfunctions_activity_worker/activity_worker.py [hunk 5]
```diff
@@ -68,8 +74,6 @@ def perform_task(self):
         except (Exception, KeyboardInterrupt) as error:
             *_, raw_traceback = sys.exc_info()
             formatted_traceback = traceback.format_tb(raw_traceback)
-
-            print(f"{self.activity_name} failed!")
```
Comment (MaxwellGBrown) [inline stepfunctions_activity_worker/activity_worker.py]: This print got lost ☹️

File: tests/test_activity_worker.py [hunk 6]
```diff
@@ -87,12 +87,15 @@ def test_activity_worker_gets_tasks(activity_worker_kwargs, tasks):
...
     """
     mock_client = activity_worker_kwargs["client"]
-    mock_client.get_activity_task.side_effect = tasks
+    mock_client.get_activity_task.side_effect = [
+        *tasks, KeyboardInterrupt("This is a keyboard interrupt")
+    ]
 
     worker = ActivityWorker(**activity_worker_kwargs)
-    worker()
+    worker.listen()
```
Comment (MaxwellGBrown) [inline tests/test_activity_worker.py]: Can we revert these to just testing the basic "process a single task unit" and do a separate test for listen making sure the thing happens?

File: stepfunctions_activity_worker/activity_worker.py (hunk 5 above)
Comment (ryegye24) [reply stepfunctions_activity_worker/activity_worker.py]: done

File: tests/test_activity_worker.py (hunk 6 above)
Comment (ryegye24) [reply tests/test_activity_worker.py]: done

**Pull request merged**